src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -20,6 +20,10 @@ interface SearchResult {
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
+
+  const normalizedSearchQuery = useMemo(() => searchQuery.toLowerCase(), [
```

Comment:
I think this is unnecessary because the state will change only when the setter is called. This memo will use unnecessary resources to cache the value and I think we are overusing useMemo in this case. Feel free to change this if you think I'm right, but don't take this as a change request. More like a note for everyone.

Reply:
Agreed with @denisdimitrov , the changes made here to add `normalizedSearchQuery` as it's own memo'd item are unnecessary.

If the `searchQuery.toLowerCase()` is still desired, I would just do that in the `useEffect` block below where it is used. But I would still keep it as it's own `const` there, rather than what the original code seemed to do where it was lowercasing it at each individual location it was used.